include/onnxruntime/core/framework/op_kernel_info.h

```diff
@@ -60,6 +63,7 @@ class OpKernelInfo : public OpNodeProtoHelper<ProtoHelperNodeContext> {
   const OrtValueNameIdxMap& ort_value_name_idx_map_;
   const DataTransferManager& data_transfer_mgr_;
   ProtoHelperNodeContext proto_helper_context_;
+  const std::map<OrtDevice, AllocatorPtr>& allocators_;
```

Comment:
here the allocators_ will be initialized directly from allocators in session state, so they are using the same type (std::map)

The reasons why we use map instead of unordered_map in session state are:
1. there was a map of allocators in session state so I just follow the convention
2. If we use unordered_map, we need to add more code in the key object (OrtDevice) to make it hashable
3. I understand unordered_map is more efficient than map, but considering there won't be many allocators in this map, there won't be much difference in terms of perf